.dockerignore

```diff
@@ -5,3 +5,6 @@
 !gadget-container
 !pkg
 !tools
+!Makefile
+!*.mk
+!.git
```

Comment:
Could we use a different dockerignore file for the kubectl-gadget image? Otherwise, the addition of `.git` would invalidate the cache too often for the main image.

How to use a different dockerignore: https://github.com/moby/moby/issues/12886#issuecomment-480575928

Reply:
Updated. I left the rules for `Makefile` and `*.mk` in case BuildKit is not available.